### ISSUE TITLE
[NEW] Ошибочная установка реквизитов формы во внеконтекстных серверных методах формы

### ISSUE THREAD

**Issue by artbear** [1c-syntax/bsl-language-server#1267]
## Описание проблемы, ошибки, которую надо диагностировать
Разработчики допускают ошибки, используя внеконтекстные серверы методы, внутри которых изменяют реквизиты управляемой формы.

Например, в примере ниже `ПолеКартинки ` - это реквизит упр.формы
и в таком варианте код ниже просто присваивает код неиспользуемой переменной
```bsl
&наСервереБезКонтекста
Процедура ПолучитьКартинку(Знач Картинка)

  ПолеКартинки = ПоместитьВоВременноеХранилище(Картинка);

КонецПроцедуры
```

правильно заменить директиву компиляции на `&наСервере`

### Тип [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [x] :ant: Ошибка
* [ ] :cop: Уязвимость
* [ ] :guardsman: Потенциальная уязвимость
* [ ] :poop: Качество кода
* [ ] :trollface: Другое

### Важность [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [x] :broken_heart: Блокирующая / Blocker
* [ ] :heart: Критическая / Critical
* [ ] :yellow_heart: Важная / Major
* [ ] :blue_heart: Незначительная / Minor
* [ ] :green_heart: Информационная / Info
* [ ] :revolving_hearts: Другое

### Тэги [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTag)

* [ ] `STANDARD` - "Нарушение стандартов 1С"
* [ ] `LOCKINOS` - "Не будет работать в другой ОС"
* [ ] `SQL` - "Проблема с запросом"
* [ ] `PERFORMANCE` - "Проблема производительности"
* [ ] `BRAINOVERLOAD` - "Непонятный код"
* [ ] `BADPRACTICE` - "Плохая практика программирования"
* [ ] `CLUMSY` - "Излишние действия"
* [ ] `DESIGN` - "Ошибка в проектировании"
* [x] `SUSPICIOUS` - "Подозрительный код"
* [x] `UNPREDICTABLE` - "Непредсказуемо работающий код"
* [ ] `DEPRECATED` - "Устаревшая функциональность"
* [x] `ERROR` - "Ошибочная конструкция"
* [ ] `LOCALIZE` - "Проблемы локализации"

### Время на исправление (минут)

3 минуты

**Comment by zeegin** [comment]
Кажтся тут надо ругаться на то, что появилась неиспользуемая переменная а не на то что контекст неверный. Разработчик должен сам решить использовать ли ему переменную так как он хочет или сменить контекст.

**Comment by artbear** [comment]
Нет, в данном случае явное пересечение имени локальной переменной с существующим реквизитом формы. очень часто такое пересечение - явная ошибка.
а проверка на неиспользуемую переменную может быть очень легко пропущена, особенно, если разработчик будет не очень внимателен, и баг будет выявлен только в продуктиве.

и при разборе замечаний лучше явно увидеть возможный баг и принять решение, чем получить невнятное сообщение о переменной.

/cc @zeegin

**Comment by zeegin** [comment]
Я считаю что не надо делать диагностики, который не находят гарантированно проблему. Я не хочу разбирать кучу результаов проверки на потенциальные баги. Я хочу видеть реальные проблемы. Неправильная директива в данном случае это подсказка системы о том как можно исправить проблему с неиспользуемой переменной, а не проблема которую можно решить только сменой директивы. Если система стат анализа начинает диктовать как именно исправлять код, который можно исправить по-разному - это плохо.
Проверка на неиспользуемую переменную в данном случае гарантированно выявит баг в указанном тобой примере.
Без других примеров в который ниодна другая диагностика не выявит баг считаю делать диагностику не нужно.

Правильно - это сделать хорошую диагностику на неиспользуемую переменную и в случае если действительно это внеконтекст и есть реквизит с таким именем то добавлять кодэкшн с квикфиксом. Но опять же - при таком действии может быть привнесено больше проблем чем решено. Например может начать проявляться ошибка использовать контекстного вызова из обработчика события при активизации строки и т.п.
